### PR TITLE
Clean words

### PR DESCRIPTION
**Pull request by anyarlene**

### CODE REVIEW

File: script/clean.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+with open('amajambo_cleanedV1.csv', 'w', newline='') as f:
+  csv_out = csv.writer(f)
+  csv_out.writerows([words_list[index]] for index in range(0, len(words_list)))
+
+#count number of words in our list
+def get_number_of_elements(list):
+    count = 0
+    for element in list:
+        count += 1
+    return count
```
Comment (lionel-k) [inline script/clean.py:44]: I think you can even simplify this method like this? What do you think ?

```suggestion
def  get_number_of_elements(list):
    len(list)
```

Maybe then replace `get_number_of_elements` with just `len`

File: script/clean.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+print("Number of elements in the entire list: ", get_number_of_elements(words_list))
```
Comment (lionel-k) [inline script/clean.py:46]:
```suggestion
print("Number of elements in the entire list: ", len(words_list))
```

File: script/clean.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+def five_letter(words):
+    new_wordlist = []
+    for item in words:
+        if len(item) == 5:
+            new_wordlist += [item]
+    return new_wordlist 
+
+res = five_letter(words_list)
+print(res)
+print("Number of elements in our new list of 5 letter words: ", get_number_of_elements(res))
```
Comment (lionel-k) [inline script/clean.py:62]:
```suggestion
print("Number of elements in our new list of 5 letter words: ", len(res))
```

File: script/clean.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+bad_chars = ['!', "-"]
+
+for i in range(len(words_list)):
+    # remove ascents
+    words_list[i] = unidecode.unidecode(words_list[i])
+    words_list[i] = ''.join((filter(lambda i: i not in bad_chars, words_list[i])))
+
+#save all clean words in csv file
+with open('amajambo_cleanedV1.csv', 'w', newline='') as f:
```
Comment (lionel-k) [inline script/clean.py]: You can put the filenames of the output at the beginning of the file. Like this they are easy to change/see.

File: script/clean.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    for item in words:
+        if len(item) == 5:
+            new_wordlist += [item]
+    return new_wordlist 
+
+res = five_letter(words_list)
+print(res)
+print("Number of elements in our new list of 5 letter words: ", get_number_of_elements(res))
+#save all clean words in csv file
+with open('amajambo_5letters.csv', 'w', newline='') as f:
```
Comment (lionel-k) [inline script/clean.py]: Same here. Prefer have the name of the output filename in a constant at the beginning of the script

**Review by lionel-k — APPROVED** [review]
Good job !! 😍😍 🚀

File: script/clean.py (hunk 1 above)
Comment (anyarlene) [reply script/clean.py:44]: I'll keep this as it is at the moment. it's not working.

Comment (lionel-k) [reply script/clean.py:44]: What is the error you have ? I would be surprised that we need to do so much gymnastic to compute the size of a list in Python

File: script/clean.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+rows = []
+for row in csvreader:
+    rows.append(row)
+
+flat_list = []
+for sublist in rows:
+    for item in sublist:
+        flat_list.append(item)
+
+words_list = [item.upper() for item in flat_list]
```
Comment (lionel-k) [inline script/clean.py]: At the end I realised it is easier to play with lowercase words can you please change this as well ?

```suggestion
words_list = [item.lower() for item in flat_list]
```

File: script/clean.py (hunk 1 above)
Comment (anyarlene) [reply script/clean.py:44]: returns nothing if I try your approach

**Pull request merged**